subprojects/gradle/runner/device-provider/src/main/kotlin/com/avito/android/runner/devices/internal/kubernetes/KubernetesApi.kt

```
@@ -75,20 +76,25 @@ internal interface KubernetesApi {
             count: Int
         ) {
             logger.debug("waitForDeploymentCreationDone name=$deploymentName count=$count")
-            val isDeploymentDone = waitForCondition(
+            waitForCondition(
                 logger = logger,
                 conditionName = "Deployment $deploymentName deployed"
             ) {
                 getPods(
                     deploymentName = deploymentName
-                ).size == count
-            }
-            if (!isDeploymentDone) {
-                throw RuntimeException("Can't create deployment: $deploymentName")
+                ).flatMap { pods ->
+                    if (pods.size == count) {
+                        Result.Success(pods)
+                    } else {
+                        Result.Failure(IllegalStateException("Pods count is ${pods.size} but expected $count"))
+                    }
+                }
+            }.onFailure {
+                throw IllegalStateException("Can't create deployment: $deploymentName", it)
```

Now we don't swallow the original error